### PR TITLE
Fix goroutine leak caused by connection error during IDLE

### PR DESCRIPTION
**Pull request by fastred**
Hi, thanks for creating go-imap. I’m really enjoying working with it and learning Golang in the process.

Using pprof I found a dangling goroutine when idle is ended due to lost connection. Closing `stopOrRestart` in that case fixes the issue.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -70,6 +70,7 @@ func (c *Client) Idle(stop <-chan struct{}) error {
 			close(stopOrRestart)
 			return <-done
 		case err := <-done:
+			close(stopOrRestart)
```
Comment (emersion) [inline client.go:73]: Hmm, but this should only happen when `c.idle` has returned right?

Comment (fastred) [reply client.go:73]: Yeah, sorry for not providing a clear explanation. If `stopOrRestart` isn't closed here, a goroutine started in `Response` continues waiting for `r.Stop` and never ends: https://github.com/emersion/go-imap-idle/blob/2704abd7050ed7f2143753554ee23affdf847bd9/response.go#L30-L33

Comment (emersion) [reply client.go:73]: Ah, I see. And it doesn't really make sense to close it in `idle` because we're creating the channel.

**Review by emersion — APPROVED** [review]
Good catch, thanks!

**Pull request merged**